frontend/packages/data-portal/app/queries/useDatasetsForDeposition.ts

```diff
@@ -0,0 +1,92 @@
+import { useMemo } from 'react'
+
+import {
+  createEmptyResponse,
+  useDepositionQuery,
+} from 'app/hooks/useDepositionQuery'
+import type {
+  DataContentsType,
+  DepositionDatasetsResponse,
+} from 'app/types/deposition-queries'
+import { isDefined } from 'app/utils/nullish'
+
+/**
+ * Hook to fetch datasets for a deposition using React Query + API route.
+ *
+ * This uses a server-side API route to avoid CORS issues with direct GraphQL calls,
+ * leveraging React Query for caching and state management.
```

Comment:
This doesn't seem crazy to do, but we could also tweak the BE server to put in the appropriate CORS headers. Shouldn't be too hard to do. But I can also see this approach being reasonable in-and-of-itself, not simply a workaround.

If we ever want to make the BE be directly callable by the FE and deal with the CORS headers directly, reach out in the future and we can work on it.